src/document/sidebar.ml

```diff
@@ -18,27 +18,33 @@ let sidebar_toc_entry id content =
 module Toc : sig
   type t
 
-  val of_lang : Odoc_index.Page_hierarchy.t -> t
+  val of_page_hierarchy : Odoc_index.Page_hierarchy.t -> t
 
   val to_sidebar :
     ?fallback:string -> (Url.t * Inline.one -> Block.one) -> t -> Block.t
 end = struct
   type t = (Url.t * Inline.one) option Tree.t
 
-  let of_lang (dir : Odoc_index.Page_hierarchy.t) =
-    Tree.map dir ~f:(function
-      | None -> None
-      | Some (parent_id, title) ->
-          let path =
-            match
-              Url.from_identifier ~stop_before:false (parent_id :> Id.t)
-            with
-            | Ok r -> r
-            | Error _ -> assert false
-            (* This error case should never happen since [stop_before] is false, and even less since it's a page id *)
-          in
-          let content = Comment.link_content title in
-          Some (path, sidebar_toc_entry parent_id content))
+  let of_page_hierarchy (dir : Odoc_index.Page_hierarchy.t) =
+    let f index =
+      let payload =
```

Comment:
What's with the extra `let payload = ... in payload`?

Reply:
Fixed